numpy/random/src/distributions/distributions.c

```diff
@@ -957,6 +957,39 @@ RAND_INT_TYPE random_geometric(bitgen_t *bitgen_state, double p) {
   }
 }
 
+RAND_INT_TYPE random_two_sided_geometric(bitgen_t *bitgen_state, double alpha) {
+  double prob;
+  RAND_INT_TYPE sign;
+
+  prob = next_double(bitgen_state);
+
+  /* 
+   * This fractional term is the probability
+   * that the distribution returns 0.
+   */
+  if (prob < (1.0 - alpha)/(1.0 + alpha)) {
+    return 0;
+  } else {
+    /* 
+     * If the distribution does not return 0,
+     * then the result can be either + or -.
+     */
+    prob = next_double(bitgen_state);
```

Comment:
Get the next uint32 and use the MSB (or some other bit that is not one of the 4 LSBs)

Reply:
Something like 
```
sign = 2 * (int32_t)(next_uint32(bitgen_state) >> 31) - 1
```